Cargo.toml

```
@@ -5,20 +5,26 @@ name = "twilight-http-proxy"
 version = "0.1.0"
 
 [dependencies]
-twilight-http = { version = "0.5", default-features = false, features = ["rustls-webpki-roots"] }
-hyper = { version = "0.14", features = ["tcp", "server", "http1", "http2"] }
+dashmap = "4.0"
 http = "0.2"
+hyper = { version = "0.14", features = ["tcp", "server", "http1", "http2"] }
+tokio = { version = "1.0", features = ["rt", "rt-multi-thread", "macros"] }
 tracing = "0.1"
 tracing-subscriber = { version = "0.2", features = ["fmt", "registry"] }
 tracing-log = "0.1"
-snafu = "0.6"
-tokio = { version = "1.0", features = ["rt", "rt-multi-thread", "macros"] }
+twilight-http = { version = "0.5", default-features = false, features = ["rustls-webpki-roots"] }
 
 # Only used by the `expose-metrics` feature.
-metrics = { version = "0.16", optional = true }
-metrics-exporter-prometheus = { version = "0.5", optional = true }
+metrics = { version = "0.17", optional = true }
+metrics-exporter-prometheus = { version = "0.6", optional = true }
 lazy_static = { version = "1.4", optional = true }
 
 [features]
 default = []
 expose-metrics = ["metrics", "metrics-exporter-prometheus", "lazy_static"]
+
+[profile.release]
+codegen-units = 1
+incremental = false
+lto = true
+opt-level = 3
```

opt-level & incremental already default to those values. `panic = 'abort'` can be set to further reduce code size.
```suggestion
codegen-units = 1
lto = true
panic = 'abort'
```

---

I usually add `panic = 'abort'` for my own projects too but wasn't 100% sure if that's desired here since the interaction with tokio might be a bit awkward - I don't know what happens if one task panics, e.g. the HTTP client

---

Panics in `tokio::spawn` return `JoinError` (https://docs.rs/tokio/1.9.0/tokio/task/struct.JoinHandle.html#impl-Future)